addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -985,6 +985,13 @@ options.registry.WebsiteFieldEditor = FieldEditor.extend({
      * Set the name of the field on the label
      */
     setLabelText: function (previewMode, value, params) {
+        if (!previewMode && this._getDependencyEl() &&
+            this._recursiveFindCircular(this._getDependencyEl().closest(".s_website_form_field"))) {
+            // A circular dependency can appear if for example a field (A)
+            // depends on another (B) and the user rename "A" by "B". In this
+            // case, remove the problematic circular dependency.
+            this._deleteConditionalVisibility(this.$target[0]);
+        }
```

Comment:
Just a suggestion, though I think storing dependencyEl is probably a nice thing :D
```js
const dependencyEl = this._getDependencyEl():
if (
    !previewMode 
    && dependencyEl 
    && this._recursiveFindCircular(dependencyEl.closest(".s_website_form_field"))
 ) {
     ...
 }
 ```

Reply:
@detrouxdev , Indeed, thanks for reporting! I adapted the PR accordingly :+1: